test/unit/DCAKeep3rJob/dca-keep3r-job.spec.ts

```diff
@@ -135,4 +143,152 @@ contract('DCAKeep3rJob', () => {
+      when: 'nonce is invalid',
+      signer: () => canSign,
+      nonce: 10,
+      txFailsWith: 'SignerCannotSignWork',
+    });
+    workFailsTest({
+      when: 'signer is not allowed to sign',
+      signer: () => random,
+      txFailsWith: 'SignerCannotSignWork',
+    });
+
+    when('work is called correctly', () => {
+      let keeper: Wallet;
+      given(async () => {
+        const { v, r, s } = await getSignature({
+          signer: canSign,
+          swapper: swapper.address,
+          data: callData,
+          nonce: 0,
+          chainId,
+        });
+        keep3r.isKeeper.returns(true);
+        keeper = await wallet.generateRandom();
+        await DCAKeep3rJob.connect(keeper).work(callData, v, r, s);
+      });
+      then('keeper check was executed corrctly', () => {
+        expect(keep3r.isKeeper).to.have.been.calledOnceWith(keeper.address);
+      });
+      then('nonce is increased', async () => {
+        const { nonce } = await DCAKeep3rJob.swapperAndNonce();
+        expect(nonce).to.equal(1);
+      });
+      then('swapper is called correctly', async () => {
+        expect(swapper.swap).to.have.been.calledOnceWith(TOKEN_IN, AMOUNT_IN, TOKEN_OUT);
+      });
+      then('worked is called correctly', () => {
+        expect(keep3r.worked).to.have.been.calledOnceWith(keeper.address);
+      });
+    });
+
+    function workFailsTest({
+      when: title,
+      signer,
+      txFailsWith,
+      callerIsNotAKeeper,
+      ...call
+    }: {
+      when: string;
+      signer: () => SignerWithAddress;
+      callerIsNotAKeeper?: boolean;
+      txFailsWith: string;
+    } & Partial<Omit<OperationData, 'signer'>>) {
+      when(title, () => {
+        given(() => keep3r.isKeeper.returns(!callerIsNotAKeeper));
+        then('reverts with message', async () => {
+          const { v, r, s } = await getSignature({
+            signer: signer(),
+            swapper: swapper.address,
+            data: call.data ?? callData,
+            nonce: 0,
+            chainId,
+            ...call,
+          });
+          await behaviours.txShouldRevertWithMessage({
+            contract: DCAKeep3rJob,
+            func: 'work',
+            args: [callData, v, r, s],
+            message: txFailsWith,
+          });
+        });
+      });
+    }
+
+    const Work = [
+      { name: 'swapper', type: 'address' },
+      { name: 'data', type: 'bytes' },
+      { name: 'nonce', type: 'uint256' },
+    ];
+
+    async function getSignature(options: OperationData) {
+      const { domain, types, value } = buildWorkData(options);
+      const signature = await options.signer._signTypedData(domain, types, value);
+      return fromRpcSig(signature);
+    }
+
+    function buildWorkData(options: OperationData) {
+      return {
+        primaryType: 'Work',
+        types: { Work },
+        domain: { name: 'Mean Finance - DCA Keep3r Job', version: '1', chainId: options.chainId, verifyingContract: DCAKeep3rJob.address },
```

Comment:
Just realized while reading this... Seems sensible to make the `domain` much shorter. `MF - DCAKJ` maybe? Since every byte being hashed on-chain to check signer will make it more gas-expensive. WDYT?

Reply:
That string is hashed only during the deployment. It's not hashed on each verify, so we would only be saving a little gas on the deployment. Since that's the case, I'd prefer to make it more readable, if it's ok with you of course